bundles/http/Citrix/0b4d1a6b-30f3-4134-a354-42fa008b757e/f3b61af4-efd0-43c2-9a08-a467fc8242c1.js

```diff
@@ -0,0 +1,1029 @@
+let parser = library.load("xml2js")
+let jp = library.load("jsonpath-plus")
+const moment = library.load("moment-timezone")
+
+const createRequest = (client,session) => async (body) => {
+    try {
+        const response = await client.fetch('/wfc/XmlService', {
+            method: "POST",
+            headers: {
+                "Content-Type": "application/xml",
+                Cookie: session
+            },
+            body: body
+        });
+        const data = await response.text();
+        const result = await parser.parseStringPromise(data);
+        let status = 200
+        if (response.ok && result?.Kronos_WFC.Response[0].$.Status == 'Failure') {
+            status = 400
+            throw new Error(result.Kronos_WFC.Response[0]);
+        }
+
+        return {
+            response,
+            result,
+            status
+           // or maybe even better: `result: result.Kronos_WFC.Response[0]` ?          
+        }
+    } catch (error) {
+        throw new Error(JSON.stringify(error))
+    }
+}
+
+async function logIn(client, userName, password) {
+    let session = [];
+    const response = await client.fetch('/wfc/XmlService', {
+        method: "POST",
+        headers: {
+            "Content-Type": "application/xml",
+            "Accept": "application/xml"
+        },
+        body: `<KRONOS_WFC VERSION="1.0"><REQUEST OBJECT="SYSTEM" ACTION="LOGON" USERNAME="${userName}" PASSWORD="${password}"/></KRONOS_WFC>`
+    });
+
+    const cookies = await response.headers.get('set-cookie');
+    const data = await response.text();
+    parser.parseString(data, (err , result)=>{
+        if(err != null){
+            throw new Error(JSON.stringify(err))
+        }
+        else if(result.Kronos_WFC.Response[0].$.Status == 'Failure'){
+            throw new Error(result.Kronos_WFC.Response[0].$.Message)
+        }
+
+        session = cookies.split(',').filter(value =>{
+            return value.indexOf('JSESSIONID') === 0
+        })
+    })
+    return session[0];
+}
+
+async function syncPeople(dataStore, client, session, integrationParameters) {
+    const month = moment().format('M')
+    const start = moment().month(month-1).startOf('month').format(integrationParameters.dateFormat)
+    const end = moment().month(month).endOf('month').format(integrationParameters.dateFormat)
+    const body =  `<?xml version="1.0"?>
+    <KRONOS_WFC VERSION="1.0"><Request Action = "RunQuery">
+    <HyperfindQuery HyperfindQueryName = "All Home"  QueryDateSpan = "${start} - ${end}" VisibilityCode= "Public"  QueryPersonOrEmployee="Person" />
+    </Request>
+    </KRONOS_WFC>`
+    const request = createRequest(client,session)
+    const {response,result} = await request(body)
```

Comment:
**Kronos Does not support pagination**, so we do limit all the cache to 2 months data maximum and for Record Time Stamp and Accrual Balance we do have only 2 days data into cache.

Reply:
Ok, my only question is if in case of company with e.g. 100 000 employees the response cannot exceed 128 kB which is a limit of MAS. If you're sure it won't it's fine then.